tasks/create_http_task.rb

```diff
@@ -0,0 +1,91 @@
+# Copyright 2018 Google, Inc
+# Licensed under the Apache License, Version 2.0 (the "License")
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START cloud_tasks_appengine_create_task]
+require "google/cloud/tasks"
+
+# Create an App Engine Task
+#
+# @param [String] project_id Your Google Cloud Project ID.
+# @param [String] location_id Your Google Cloud Project Location ID.
+# @param [String] queue_id Your Google Cloud App Engine Queue ID.
+# @param [String] url The full path to sent the task request to.
+# @param [String] payload The request body of your task.
+# @param [Integer] seconds The delay, in seconds, to process your task.
+def create_http_task project_id, location_id, queue_id, url, payload: nil, seconds: nil
+  # Instantiates a client.
+  client = Google::Cloud::Tasks.new
+
+  # Construct the fully qualified queue name.
+  parent = client.queue_path project_id, location_id, queue_id
+
+  # Construct task.
+  task = {
+    http_request: {
+      http_method: "POST",
+      url:         url
+    }
+  }
+
+  # Add payload to task body.
+  if payload
+    task[:http_request][:body] = payload
+  end
+
+  # Add scheduled time to task.
+  if seconds
+    timestamp = Google::Protobuf::Timestamp.new
+    timestamp.seconds = Time.now.to_i + seconds.to_i
+    task[:schedule_time] = timestamp
+  end
+
+  # Send create task request.
+  puts "Sending task #{task}"
+  response = client.create_task parent, task
+
+  if response.name
+    puts "Created task #{response.name}"
+  end
+end
+# [END cloud_tasks_appengine_create_task]
+
+if $PROGRAM_NAME == __FILE__
+  project_id  = ARGV.shift
```

Comment:
Elsewhere in the repo, project_id is grabbed from ENV rather than as an argument.

Reply:
I.E. https://github.com/GoogleCloudPlatform/ruby-docs-samples/blob/13b453bd54baf7eed1e1d40bd6a45e9cf007f91a/storage/iam.rb#L73